code/game/objects/items/devices/binoculars.dm

```diff
@@ -335,6 +335,112 @@
 	cooldown_duration = 80
 	target_acquisition_delay = 30
 
+/obj/item/device/binoculars/range/designator/spotter
+	name = "spotter's laser designator"
+	desc = "A specially-designed laser designator, issued to USCM spotters, with two modes: target marking for CAS with IR laser and rangefinding. Ctrl + Click turf to target something. Ctrl + Click designator to stop lasing. Alt + Click designator to switch modes."
+
+	var/spotting_time = 10 SECONDS
+	var/spotting_cooldown
+	var/spotting_cooldown_delay = 5 SECONDS
+
+/obj/item/device/binoculars/range/designator/spotter/Initialize()
+	LAZYADD(actions_types, /datum/action/item_action/specialist/spotter_target)
+	return ..()
+
+/datum/action/item_action/specialist/spotter_target
+	ability_primacy = SPEC_PRIMARY_ACTION_1
+	var/minimum_laze_distance = 2
+
+/datum/action/item_action/specialist/spotter_target/New(var/mob/living/user, var/obj/item/holder)
+	..()
+	name = "Spot Target"
+	button.name = name
+	button.overlays.Cut()
+	var/image/IMG = image('icons/mob/hud/actions.dmi', button, "spotter_target")
+	button.overlays += IMG
+	var/obj/item/device/binoculars/range/designator/spotter/designator = holder_item
+	designator.spotting_cooldown = world.time
+
+/datum/action/item_action/specialist/spotter_target/action_activate()
+	if(!ishuman(owner))
+		return
+	var/mob/living/carbon/human/H = owner
+	if(H.selected_ability == src)
+		to_chat(H, "You will no longer use [name] with \
+			[H.client && H.client.prefs && H.client.prefs.toggle_prefs & TOGGLE_MIDDLE_MOUSE_CLICK ? "middle-click" : "shift-click"].")
+		button.icon_state = "template"
+		H.selected_ability = null
+	else
+		to_chat(H, "You will now use [name] with \
+			[H.client && H.client.prefs && H.client.prefs.toggle_prefs & TOGGLE_MIDDLE_MOUSE_CLICK ? "middle-click" : "shift-click"].")
+		if(H.selected_ability)
+			H.selected_ability.button.icon_state = "template"
+			H.selected_ability = null
+		button.icon_state = "template_on"
+		H.selected_ability = src
+
+/datum/action/item_action/specialist/spotter_target/can_use_action()
+	var/mob/living/carbon/human/H = owner
+	if(!(GLOB.character_traits[/datum/character_trait/skills/spotter] in H.traits))
+		to_chat(H, SPAN_WARNING("You have no idea how to use this!"))
+		return FALSE
+	if(istype(H) && !H.is_mob_incapacitated() && !H.lying && (holder_item == H.r_hand || holder_item || H.l_hand))
+		return TRUE
+
+/datum/action/item_action/specialist/spotter_target/proc/use_ability(atom/A)
```

Comment:
people in your own faction shouldn't be spottable (nor aimed shottable since the gun has iff)

Reply:
i agree with kaga

Reply:
fairs